### ISSUE TITLE
Proposal for new a11y check to disallow headings with no text content

### ISSUE THREAD

**Issue by glnster** [ember-template-lint/ember-template-lint#2114]
Currently, ember-template-lint does not prevent empty headings (h1, h2, etc.). Headings relay the structure of a webpage and provide a meaningful, hierarchical order of its content. If headings are empty or its text contents is inaccessible, this could confuse users or prevent them accessing sections of interest.

This proposes the authoring of a new rule that enforces that headings have accessible text content. Headings with no text content or content that is hidden (using `aria-hidden`) should be disallowed. This is essentially a port of JSXs A11y [heading-has-content](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/master/docs/rules/heading-has-content.md) rule.

Reference: [WCAG SC 2.4.6 Headings and Labels](https://www.w3.org/TR/UNDERSTANDING-WCAG20/navigation-mechanisms-descriptive.html)

**Comment by MelSumner** [comment]
We should probably also check for an element with the role of heading and an aria-level attribute. 

Make sure we reference the WCAG guidelines in the doc and not the jsx rule, though. 👍

**Comment by glnster** [comment]
> We should probably also check for an element with the role of heading and an aria-level attribute.
> 
> Make sure we reference the WCAG guidelines in the doc and not the jsx rule, though. 👍

Good call-outs! Thank you. Linked to the guideline to reference from.
